arch/lkl/kernel/vmlinux.lds.S

```diff
@@ -8,28 +8,47 @@ OUTPUT_FORMAT(CONFIG_OUTPUT_FORMAT)
 
 jiffies = jiffies_64;
 
+/* FIXME: symbols defined here outside individual sections don't have a final
+ * address assigned because vmlinux is linked incrementally. In the final
+ * linking step (when linked against an application) they can be rearranged away
+ * from the other adjacent symbols or sections in this file which makes them
+ * bogus.
+ *
+ * Fortunatelly, in most cases their are used for informational purposes only,
+ * e.g. printing the init/code/data size early during boot.
+ */
 SECTIONS
 {
-	__init_begin = .;
-	HEAD_TEXT_SECTION
```

Comment:
What about `HEAD_TEXT_SECTION` and `INIT_TEXT_SECTION`? Flagging this just to make sure it's not forgotten.

Reply:
INIT_TEXT_SECTION expands to 

```
#define INIT_TEXT_SECTION(inittext_align)                               \                                                     
        . = ALIGN(inittext_align);                                      \                                                     
        .init.text : AT(ADDR(.init.text) - LOAD_OFFSET) {               \                                                     
                _sinittext = .;                                         \                                                     
                INIT_TEXT                                               \                                                     
                _einittext = .;                                         \                                                     
        } 
```

and we moved INIT_TEXT and the _sinittext/_einittext assignments directly into the .text section

Reply:
Ah, got it. And regarding HEAD_TEXT_SECTION which seems to expand to `KEEP(*(.head.text))` do we need it at all?

Reply:
HEAD_TEXT_SECTION is used by the arch code for early boot stuff (e.g. decompression). We don't use it. Just to make sure, I double checked it:

```
$ find -name \*.o | xargs objdump -h | grep head.text | wc -l
0
```